src/main/java/uk/ac/manchester/beehive/tornado/plugins/dynamicInspection/CodeGenerator.java

```diff
@@ -123,11 +127,22 @@ private static File createFile(Project project, PsiMethod method, ArrayList<PsiM
         return javaFile;
     }
 
+    private static boolean isParameterBoxedType(PsiParameter p) {
+        return switch (p.getTypeElement().getText()) {
+            case "int" -> false;
+            case "float" -> false;
+            case "double" -> false;
+            case "long" -> false;
+            case "boolean" -> false;
+            default -> true;
+        };
+    }
+
```

Comment:
```suggestion
 private static boolean isParameterBoxedType(PsiParameter p) {
    return switch (p.getTypeElement().getText()) {
        case "int", "float", "double", "long", "boolean" -> false;
        default -> true;
    };
}
```

This can be simplified with enhanced switch